cedar-policy-core/src/ast/entity.rs

```diff
@@ -242,38 +372,79 @@ impl<'a> arbitrary::Arbitrary<'a> for EntityUID {
 /// (see [#884](https://github.com/cedar-policy/cedar/issues/884)).
 /// To get an escaped representation, use `.escaped()`.
 /// To get an unescaped representation, use `.as_ref()`.
-#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Hash, PartialOrd, Ord)]
-pub struct Eid(SmolStr);
+#[derive(PartialEq, Eq, Debug, Clone, Hash, PartialOrd, Ord)]
+pub enum Eid {
+    /// Actual Eid
+    Eid(SmolStr),
+    #[cfg(feature = "tolerant-ast")]
+    /// Represents an Eid of an entity that failed to parse
+    ErrorEid,
+}
+
+impl<'de> Deserialize<'de> for Eid {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = String::deserialize(deserializer)?;
+        Ok(Eid::Eid(SmolStr::from(value)))
+    }
+}
+
+impl Serialize for Eid {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            Eid::Eid(s) => s.serialize(serializer),
+            #[cfg(feature = "tolerant-ast")]
+            Eid::ErrorEid => serializer.serialize_str("Eid::Error"),
+        }
+    }
+}
 
 impl Eid {
     /// Construct an Eid
     pub fn new(eid: impl Into<SmolStr>) -> Self {
-        Eid(eid.into())
+        Eid::Eid(eid.into())
     }
 
     /// Get the contents of the `Eid` as an escaped string
     pub fn escaped(&self) -> SmolStr {
-        self.0.escape_debug().collect()
+        match self {
+            Eid::Eid(smol_str) => smol_str.escape_debug().collect(),
+            #[cfg(feature = "tolerant-ast")]
+            Eid::ErrorEid => "Eid::Error".into(),
```

Comment:
We should put the string in a constant and use it everywhere (`serialize`, `as_ref`, etc.)

Reply:
Will do